src/analysis/normed_space/box_subadditive.lean

```diff
@@ -0,0 +1,541 @@
+/-
+Copyright (c) 2020 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Yury G. Kudryashov
+-/
+import data.real.ennreal
+import topology.metric_space.basic
+import linear_algebra.affine_space.ordered
+import analysis.normed_space.add_torsor
+import analysis.specific_limits
+import analysis.asymptotics
+
+/-!
+# Sub/sup-additive functions on boxes
+
+Let `s` be a set in `ι → ℝ`. A subbox of `s` is a product of closed intervals which is included
+by `s`.
+
+A function `f : (ι → ℝ) → (ι → ℝ) → M` is called
+`box_subadditive_on`/`box_additive_on`/`box_supadditive_on` a set `s` if for any `l u : ι → ℝ`, `l ≤
```

Comment:
The usual vocabulary is `subfoo` and `superfoo` (like for martingales, multiplicativity and so on). It is only two more characters, so I would use `box_superadditive_on` here.